Evolutionary_loop/Loop_Parts/Part_B/part_b_job1.py

```diff
@@ -0,0 +1,229 @@
+    sp.run(f'[ -f {sim_path} ] && > {sim_path}', shell=True)
+    
+    # Begin writing some run-specific variables to the xmacro
+    with open (sim_path, "w+") as f:
+        f.write(f'var NPOP = {npop};\n'
+                f'var indiv = {indiv};\n'
+                f'//Factor of {geo_factor} frequency\n'
+                'var freq = [')
+        f.write(', '.join( "%.2f"%(freq*geo_factor) for freq in freq_list) )
+        f.write('] \n')
+    
+    if (gen == 0 and indiv == 1):
+        with open (sim_path,"a") as f:
+            f.write('if(indiv==1){\n'
+                    f'App.saveCurrentProjectAs(\"{working_dir}/Run_Outputs/'
+                    f'{run_name}/{run_name}\");\n'
+                    '}\n')
+
+    os.umask(0)                # umask(0) needed to ensure chmod works properly.
+    os.chmod(sim_path, 0o775 ) # This is the same as Bash chmod 775 ${sim_path}.
+
+
+## CAT SECTION
+
+    # The rest of simulation_PEC.xmacro is built from two skeleton text files.
+    if curved == 0:                            
+        if nsections == 1:                     # straight-side symmetric bicone
+            sp.run(f'cat {xmacros_dir}/simulationPECmacroskeleton_GPU.txt'
+                   f'>> {sim_path}', shell=True)
+            sp.run(f'cat {xmacros_dir}/simulationPECmacroskeleton2_GPU.txt'
+                   f'>> {sim_path}', shell=True)
+
+        else:                                  # straight-side asymmetric bicone
+            sp.run(f'cat {xmacros_dir}/simulationPECmacroskeleton_Sep.txt'
+                   f'>> {sim_path}', shell=True)
+            sp.run(f'cat {xmacros_dir}/simulationPECmacroskeleton2_Sep.txt'
+                   f'>> {sim_path}', shell=True)
+    else:                                      # curved-side bicone
+        sp.run(f'cat {xmacros_dir}/simulationPECmacroskeleton_curved.txt'
+               f'>> {sim_path}', shell=True)
+        sp.run(f'cat {xmacros_dir}/simulationPECmacroskeleton2_curved.txt'
+               f'>> {sim_path}', shell=True)
+
+
+## SED-TION (Ha! Get it? Like...section?)
+
+    # sed swaps out the hardcoded word "fileDirectory" inside the xmacro.
+    # option "i" of sed: swap "in-place" instead of outputting to terminal.
+    # The empty quotes after sed -i are for MacOS compatibility:
+    # https://stackoverflow.com/questions/12272065/sed-undefined-label-on-macos
+    sp.run(f'sed -i "" "s|fileDirectory|{working_dir}/Generation_Data|" '
+           f'{sim_path}', shell=True)
+
+    # Change gridsize by the same factor used for changing antenna size.
+    xmacro_skeleton_default_gridsize = 0.1
+    grid_size = "%.6f" % (xmacro_skeleton_default_gridsize/geo_factor)
+
+    sp.run(f'sed -i "" "s/var gridSize = 0.1;/var gridSize = {grid_size};/" '
+           f'{sim_path}', shell=True)
+
+
+## XF SECTION
+
+    os.chmod(xmacros_dir, 0o775)
+    # load XFdtd on OSC
+    p0 = sp.run(f'module load xfdtd/7.9.2.2', shell=True, capture_output=True)
```

Comment:
How does this work?  I was under the impression that this did something that was used later in the script, but if you execute it in a subprocess, that gets destroyed immediately then what happens next will be unaffected unless it changes something globally in the environment.

Reply:
Yeah this absolutely doesn't actually work and is in the TODO list at the top of the script.
I'll have to test on OSC in the future. We probably have to run `module load xfdtd` and `xfdtd {xf_proj}...` all in the same `subprocess`.